### ISSUE TITLE
Split up SNTemp input_output into separate input & output 

### ISSUE THREAD

**Issue by jzwart** [USGS-R/delaware-model-prep#38]
Currently, we have `uncal_sntemp_input_output` that contains variables used to drive the model (e.g. precipitation, air temperature) alongside sntemp output (e.g. water temperature, discharge, etc...). Splitting these up into two separate files (e.g. `sntemp_input`, `sntemp_output`) would be easier for ML workflows. 

currently producing output in [delaware-water-temp workflow ](https://github.com/jzwart/delaware-water-temp/blob/6cf7768375355ce6e3d5145ad4ea2dc06a81a355/4_model_for_PGDL.yml#L42-L46)

**Comment by limnoliver** [comment]
@jzwart -- This is one potential task I thought Hayley could tackle early on, which is to move some of the run(SNTEMP) and extract(SNTEMP) functions to this pipeline. That way, she can get comfortable running and analyzing SNTEMP output.

**Comment by jzwart** [comment]
Yeah that would be great. I can walk her through some of the functions, and will also need to comment the functions a bit better than I have right now.

**Comment by jzwart** [comment]
While getting tmin / tmax for drivers, I'm also splitting out all drivers into one file (`sntemp_input.feather`) and sntemp output into one file. I'm doing an area-weighted average of tmin, tmax, and prcp for the drivers and am getting some different patterns from comparable sntemp output. For example, we have been storing `seg_rain` which is the `Area-weighted average rainfall for each segment from HRUs contributing flow to the segment` . When I compare sntemp-calculated `seg_rain` to what I have calculated `seg_prcp`, there are a few rain events where `seg_prcp` is higher than `seg_rain`. 
![image](https://user-images.githubusercontent.com/4825431/81631604-37ff8b00-93d6-11ea-8cd1-f5805780939b.png)
Across all days, ~24% of the days have different `seg_prcp` and `seg_rain`. However, most of these differences are small as ~2% of days have a difference greater than 1 mm/day precipitation. 
![image](https://user-images.githubusercontent.com/4825431/81632673-e0165380-93d8-11ea-9e93-40a245b90f43.png)

**Comment by limnoliver** [comment]
Are these anomalies at specific reaches, or distributed across the network? The `...HRUs contributing flow to the segment` made me think maybe some HRUs are flagged as non-contributing, and therefore excluding, making your estimate higher?

**Comment by jzwart** [comment]
The figures above are from the 42 segment subbasin that Xiaowei has been working on. All segments in this subbasin have at least one HRU contributing based on the parameter file used by PRMS-SNTemp. There isn't a clear pattern in the difference in calculated precip (for precip difference greater than 1 mm) for the segments based on contributing HRUs
![image](https://user-images.githubusercontent.com/4825431/81691988-be928780-942b-11ea-8e79-289577cc9882.png)

**Comment by jzwart** [comment]
Main function that I used for calculating area-weighted average of segment driver input is [here](https://github.com/jzwart/delaware-water-temp/blob/tmin_tmax_drivers/4_model_for_PGDL/src/get_segment_driver.R). 
Main block of code that SNTemp uses for calculating area-weighted average of segment driver input is [here](https://github.com/jzwart/prms/blob/f20c0bbb072222a31215dfe0cd45e35f186e8210/prms/stream_temp.f90#L769-L832).

**Comment by jsadler2** [comment]
I'm thinking that this wouldn't be that helpful after all. I mean it'd be fine either way, but I don't see a major benefit in splitting them like I thought I would have. So as long as they're together, maybe we leave them together, unless someone else feels strongly that it'd be easier to have the input and output as separate files.
